qemu/tests/sev_dhcert_boot.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+from avocado.utils import process
+from virttest import error_context
+from virttest import env_process
+from virttest import utils_package
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu sev basic test on Milan and above host:
+    1. Check host sev capability
+    2. Generate dhcert and session files
+    3. Boot sev VM with dhcert
+    4. Verify sev enabled
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Start sev test", test.log.info)
+    timeout = params.get_numeric("login_timeout", 240)
+
+    output = process.getoutput(params["sev_host_check"], shell=True)
+    if output not in params["module_status"]:
+        test.cancel("Host sev capability check fail.")
+
+    sev_tool_pkg = params.get("sev_tool_pkg")
+    s, o = process.getstatusoutput("rpm -qa | grep %s" % sev_tool_pkg,
+                                   shell=True)
+    if s != 0:
+        install_status = utils_package.package_install(sev_tool_pkg)
+        if not install_status:
+            test.cancel("Failed to install %s." % sev_tool_pkg)
+
+    vm_name = params["main_vm"]
+    try:
+        process.system_output("sevctl export --full vm.chain", shell=True)
+        process.system_output("sevctl session --name " + vm_name +
+                              " vm.chain " + params["vm_sev_policy"],
+                              shell=True)
+        params["vm_sev_dh_cert_file"] = os.path.abspath("%s_godh.b64"
+                                                        % vm_name)
+        params["vm_sev_session_file"] = os.path.abspath("%s_session.b64"
+                                                        % vm_name)
+    except Exception as e:
+        test.fail("Insert guest dhcert and session blob failed, %s" % e)
+
+    env_process.preprocess_vm(test, params, env, vm_name)
+    vm = env.get_vm(vm_name)
+    vm.create()
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    vm.monitor.query_sev_launch_measure()
+    try:
+        session.cmd_output(params["sev_guest_check"], timeout=240)
+    except Exception as e:
+        test.fail("Guest sev verify fail: %s" % e)
+    session.close()
+    vm.destroy()
```

Comment:
Add them in finally.
Some people like to let VT do the close and destroy work, some prefer to remove them in finally clause because they created/started them in case, either is OK

Reply:
Thanks, adding  vm.destroy and session.close and file delete on the finally clause.